newrelic/resource_newrelic_synthetics_private_location.go

```diff
@@ -114,13 +115,15 @@ func resourceNewRelicSyntheticsPrivateLocationRead(ctx context.Context, d *schem
 
 	resp, err := client.Entities.GetEntityWithContext(ctx, guid)
 	if err != nil {
-		if _, ok := err.(*errors.NotFound); ok {
-			d.SetId("")
-			return nil
+		if err.Error() == "Argument \"guid\" has invalid value $guid." {
+			return diag.FromErr(fmt.Errorf("invalid GUID"))
 		}
-
 		return diag.FromErr(err)
 	}
+	if _, ok := err.(*errors.NotFound); ok {
```

Comment:
Why did you move this code out of the err != nil if statement?

Reply:
I would keep it for now. The goal is to get the errors.NotFound working again at some point.

Reply:
done